dotnet/src/Connectors/Connectors.Amazon.UnitTests/Services/BedrockChatCompletionServiceTests.cs

```diff
@@ -0,0 +1,590 @@
+        // Act
+        List<StreamingChatMessageContent> output = new List<StreamingChatMessageContent>();
+        var result = service.GetStreamingChatMessageContentsAsync(chatHistory).ConfigureAwait(true);
+
+        // Assert
+        await foreach (var item in result)
+        {
+            Assert.NotNull(item);
+            Assert.NotNull(item.Content);
+            Assert.NotNull(item.Role);
+            output.Add(item);
+        }
+        Assert.NotNull(output);
+        Assert.NotNull(service.GetModelId());
+        Assert.NotNull(service.Attributes);
+    }
+
+    private static ChatHistory CreateSampleChatHistory()
+    {
+        var chatHistory = new ChatHistory();
+        chatHistory.AddUserMessage("Hello");
+        chatHistory.AddAssistantMessage("Hi");
+        chatHistory.AddUserMessage("How are you?");
+        return chatHistory;
+    }
+    /// <summary>
+    /// Checks that the prompt execution settings are correctly registered for the chat completion call.
+    /// </summary>
+    [Fact]
+    public async Task TitanGetChatMessageContentsAsyncShouldReturnChatMessageWithPromptExecutionSettingsAsync()
+    {
+        // Arrange
+        string modelId = "amazon.titan-text-lite-v1";
+        var mockBedrockApi = new Mock<IAmazonBedrockRuntime>();
+        var executionSettings = new PromptExecutionSettings()
+        {
+            ModelId = modelId,
+            ExtensionData = new Dictionary<string, object>()
+            {
+                { "temperature", 0.7f },
+                { "topP", 0.9f },
+                { "maxTokenCount", 512 }
+            }
+        };
+        mockBedrockApi.Setup(m => m.ConverseAsync(It.IsAny<ConverseRequest>(), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new ConverseResponse
+            {
+                Output = new ConverseOutput
+                {
+                    Message = new Message
+                    {
+                        Role = ConversationRole.Assistant,
+                        Content = new List<ContentBlock> { new ContentBlock { Text = "I'm doing well." } }
+                    }
+                },
+                Metrics = new ConverseMetrics(),
+                StopReason = StopReason.Max_tokens,
+                Usage = new TokenUsage()
+            });
+        var service = new BedrockChatCompletionService(modelId, mockBedrockApi.Object);
+        var chatHistory = CreateSampleChatHistory();
+
+        // Act
+        var result = await service.GetChatMessageContentsAsync(chatHistory, executionSettings).ConfigureAwait(true);
+
+        // Assert
+        ConverseRequest converseRequest = new ConverseRequest();
+        var invocation = mockBedrockApi.Invocations
+            .Where(i => i.Method.Name == "ConverseAsync")
+            .SingleOrDefault(i => i.Arguments.Count > 0 && i.Arguments[0] is ConverseRequest);
+        if (invocation != null)
+        {
+            converseRequest = (ConverseRequest)invocation.Arguments[0];
+        }
```

Comment:
I'd suggest writing this like below. It removes the unnecessary call to create a `new ConverseRequest();` which is just then ignored, when really you're just trying to prevent a null reference case, which can be done with `Assert.NotNull`

```suggestion
        var invocation = mockBedrockApi.Invocations
            .Where(i => i.Method.Name == "ConverseAsync")
            .SingleOrDefault(i => i.Arguments.Count > 0 && i.Arguments[0] is ConverseRequest);
        Assert.NotNull(invocation);
        ConverseRequest converseRequest = (ConverseRequest)invocation.Arguments[0];
```